When the Ethernet controller is powered down and someone wants to 
access the mdio bus like the witch driver (b53) the system crashed if 
PCI_D3hot was set before. This patch deactivates this power sawing mode 
when a switch driver is in use.

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -4221,7 +4221,8 @@ static void tg3_power_down(struct tg3 *t
 	tg3_power_down_prepare(tp);
 
 	pci_wake_from_d3(tp->pdev, tg3_flag(tp, WOL_ENABLE));
-	pci_set_power_state(tp->pdev, PCI_D3hot);
+	if (!tg3_flag(tp, ROBOSWITCH))
+		pci_set_power_state(tp->pdev, PCI_D3hot);
 }
 
 static void tg3_aux_stat_to_speed_duplex(struct tg3 *tp, u32 val, u16 *speed, u8 *duplex)
